js/src/common/Translator.ts

```diff
@@ -4,34 +4,44 @@ import username from './helpers/username';
 import extract from './utils/extract';
 
 export default class Translator {
-  constructor() {
-    /**
-     * A map of translation keys to their translated values.
-     *
-     * @type {Object}
-     * @public
-     */
-    this.translations = {};
+  /**
+   * A map of translation keys to their translated values.
+   */
+  translations = {};
+
+  /**
+   * The underlying ICU MessageFormatter util.
+   *
+   * @ts-ignore
+   */
+  protected formatter = new RichMessageFormatter(null, this.formatterTypeHandlers(), mithrilRichHandler);
+
+  setLocale(locale: string) {
+    this.formatter.locale = locale;
+  }
 
-    this.formatter = new RichMessageFormatter(null, this.formatterTypeHandlers(), mithrilRichHandler);
+  addTranslations(translations: { [key: string]: string }) {
+    Object.assign(this.translations, translations);
   }
 
-  formatterTypeHandlers() {
+  /**
+   * An extensible entrypoint for extenders to register type handlers for translations.
+   */
+  protected formatterTypeHandlers() {
     return {
       plural: pluralTypeHandler,
       select: selectTypeHandler,
     };
   }
 
-  setLocale(locale) {
-    this.formatter.locale = locale;
-  }
-
-  addTranslations(translations) {
-    Object.assign(this.translations, translations);
-  }
-
-  preprocessParameters(parameters) {
+  /**
+   * A temporary system to preprocess parameters.
+   * Should not be used by extensions.
+   * TODO: An extender will be added in v1.x.
+   *
+   * @internal
+   */
+  protected preprocessParameters(parameters) {
```

Comment:
See below comment about type -- same type should be used up here.

`parameters: Record<string, unknown>`?